### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by Tobinchilongo**
## Project requirements

- Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Follow SOLID principles.
- Pay special attention to the "Single-responsiblity Principle".
  - Make sure that the role of `main.rb` is only to start your code and ask the user for the initial choice of options (or to quit).
  - Extract some pieces of logic into separate class(es).

### CODE REVIEW

**Review by FrederickMih — COMMENTED** [review]
Hi Team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that shows zero(0) File changes hence I'm unable to leave a comment on this PR.

Screenshot from 2022-05-16 14-42-06

Please fix this issue so that we can easily leave inline comments on pieces of code.

Thanks :pray: 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app.rb [hunk 1]
```diff
@@ -42,85 +34,88 @@ def create_person
...
+    parent_permission = gets.chomp.downcase
+    case parent_permission
+    when 'n'
+      Student.new(age, 'classroom', name, parent_permission: false)
+      @persons << student
+      puts 'Student doesnt have parent permission, cant rent books'
+    when 'y'
+      student = Student.new(age, 'classroom', name, parent_permission: false)
+      @persons << student
+      puts 'Student created successfully'
```
Comment (Kingstalux) [inline app.rb]: - [ ] Great job so far 👍 but all requests for input and input validation should've been placed in the `main.rb` file or in a different file whose responsibilities are to request or validate input. Kindly check this [lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information on that. Right now your `app.rb` file is full of requests for input and input validation. Kindly transfer those somewhere else so as to follow SOLID and best ruby practices 👍

**Review by Kingstalux — CHANGES_REQUESTED** [review]
Hi @Tobinchilongo and @jerryowusu, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

68747470733a2f2f6d65646961302e67697068792e636f6d2f6d656469612f5754646735474252343558364e6278714a4b2f67697068792e676966

 ## Highlights 👏 
✅ Linters are passing 👍 
✅ Good commit history 👏 
✅ Console app has the correct functionality 🔥 



 ## Required Changes ♻️
_Check the comments under the review._
- [ ] Kindly add all authors to the author's section of the README file so as to make it more professional 💯 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi Team, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**